storage/aptosdb/src/pruner/pruner_manager.rs

```diff
@@ -10,25 +10,30 @@ use aptos_types::transaction::Version;
 /// The `PrunerManager` is meant to be part of a `AptosDB` instance and runs in the background to
 /// prune old data.
 ///
-/// It creates a worker thread on construction and joins it on destruction. When destructed, it
-/// quits the worker thread eagerly without waiting for all pending work to be done.
+/// If the pruner is enabled. It creates a worker thread on construction and joins it on
+/// destruction. When destructed, it quits the worker thread eagerly without waiting for
+/// all pending work to be done.
 pub trait PrunerManager: Sync {
     type Pruner: DBPruner;
 
     fn is_pruner_enabled(&self) -> bool;
 
     fn get_prune_window(&self) -> Version;
 
-    fn get_min_viable_version(&self) -> Version;
+    fn get_min_viable_version(&self) -> Version {
+        unimplemented!()
+    }
 
     fn get_min_readable_version(&self) -> Version;
 
     /// Sets pruner target version when necessary.
     fn maybe_set_pruner_target_db_version(&self, latest_version: Version);
 
-    fn set_pruner_target_db_version(&self, latest_version: Version);
+    // Only used at the end of fast sync to store the min_readable_version to db and update the
+    // in memory progress.
```

Comment:
👍

Reply:
> Beautiful. Do you plan to extract a common Manager as well? 😂

Was planning to do it, but I'm lazy, next time..